enterprise/backend/src/metabase_enterprise/metabot_v3/core.clj

```diff
@@ -18,3 +21,21 @@
   database-tables
   used-tables
   schema-sample])
+
+(defn metabot-stats
+  "Calculate total Metabot token usage over a window of the the previous UTC day 00:00-23:59"
+  []
+  (let [yesterday-utc (t/minus (t/offset-date-time (t/zone-offset "+00")) (t/days 1))]
+    {:metabot-tokens  (t2/select-one-fn :sum
+                                        [:model/MetabotMessage [:%sum.total :sum]]
+                                        {:where [:= [:cast :created_at :date] [:cast yesterday-utc :date]]})
+     :metabot-queries (t2/select-one-fn :cnt
+                                        [:model/MetabotMessage [:%count.id :cnt]]
+                                        ;; there are requests from users and responses from ai-service, in theory
+                                        ;; counting requests from users should be good enough
+                                        :role "user"
+                                        {:where [:= [:cast :created_at :date] [:cast yesterday-utc :date]]})
+     :metabot-users   (:cnt (t2/query-one {:select [[[:count [:distinct :c.user_id]] :cnt]]
+                                           :from   [[:metabot_message :m]]
+                                           :join   [[:metabot_conversation :c] [:= :c.id :m.conversation_id]]
+                                           :where [:= [:cast :m.created_at :date] [:cast yesterday-utc :date]]}))}))
```

Comment:
Could we nest all those into something like `{:metabot {:tokens .. :queries ... :users ....}}`  ?

top level is already crowded with the mandatory fields, the embedding-stats and now the metabot ones. I can see us crowding it even more, so some structure would help slice and dice those